src/capital-project/capital-project.repository.ts

```diff
@@ -47,11 +49,38 @@ export class CapitalProjectRepository {
     boroughId: string | null;
     managingAgency: string | null;
     agencyBudget: string | null;
+    commitmentsTotalMin: number | null;
+    commitmentsTotalMax: number | null;
     limit: number;
     offset: number;
   }): Promise<FindManyRepo> {
     try {
+      const commitmentsTotalByCapitalProject = this.db
+        .$with("commitmentsTotalByCapitalProject")
+        .as(
+          this.db
+            .select({
+              capitalProjectId: capitalCommitment.capitalProjectId,
+              value: sum(capitalCommitmentFund.value)
```

Comment:
It's subtle and usually won't lead to errant results. But strictly speaking, project ids are only unique within managing codes. To guarantee that we don't confuse two different projects, we'll need to select on the managing codes as well.

 ```suggestion
              managingCode: capitalCommitment.managingCode,
              capitalProjectId: capitalCommitment.capitalProjectId,
              value: sum(capitalCommitmentFund.value)
```

Reply:
I've made these updates, but I think it's a bad practice to have a table with a field named "id" and a different field as the primary key.

Reply:
fair point